runtime/kubernetes/kubernetes.go

```diff
@@ -24,20 +24,59 @@ type kubernetes struct {
 	closed chan bool
 	// client is kubernetes client
 	client client.Client
+	// namespaces which exist
+	namespaces []client.Namespace
+}
+
+// namespaceExists returns a boolean indicating if a namespace exists
+func (k *kubernetes) namespaceExists(name string) (bool, error) {
```

Comment:
Are we still making use of the "default" namespace if none exists or switching everything to a "micro" namespace?